libraries/AP_Scripting/applets/arming-checks.lua

```diff
@@ -0,0 +1,620 @@
+    if mnt1_enable then
+        local mnt1_sysid = Parameter("MNT1_SYSID_DFLT"):get() or 0
+        if mnt1_sysid > 0 and foll_sysid ~= mnt1_sysid then
+            return false
+        end
+    end
+    if mnt2_enable then
+        local mnt2_sysid = Parameter("MNT2_SYSID_DFLT"):get() or 0
+        if mnt2_sysid > 0 and foll_sysid ~= mnt2_sysid then
+            return false
+        end
+    end
+    return true
+end
+
+-- is the RTL altitude legal? This defaults to 120m (400') as set in ARM_ALT_LEGAL
+local function rtl_altitude_legal()
+    -- plane uses RTL_ALTITUDE in m
+    -- RTL_ALTITUDE will be correct for either plane or copter
+    if (RTL_ALTITUDE ~= nil and (RTL_ALTITUDE:get() or 300) > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- is the Q RTL altitude (quadplane) legal? This defaults to 120m (400') as set in ARM_ALT_LEGAL
+local function q_rtl_alt_legal()
+    -- Plane specific Parameters
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- QuadPlane specific Paraemters
+    local q_rtl_alt = Parameter("Q_RTL_ALT"):get() or 0
+    if q_rtl_alt > alt_legal_max then
+        return false
+    end
+    return true
+end
+
+-- is the RTL climb minimum legal? This defaults to 120m (400')
+local function rtl_climb_legal()
+    if (RTL_CLIMB_MIN ~= nil and RTL_CLIMB_MIN:get() > alt_legal_max) then
+        return false
+    end
+    return true
+end
+
+-- Display a message as to whether failsafe will QLand or QRTL
+local function qland_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    -- Only applies to QuadPlane
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+
+    if (not q_land_option) and (not q_RTL_option) then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
+end
+local function qrtl_warning()
+    if (Q_ENABLE:get() or 0) ~= 1 then
+        return true
+    end
+    local q_options = Parameter("Q_OPTIONS"):get() or 0
+    local q_land_option = ((q_options & (1 << 5)) > 0)
+    local q_RTL_option = ((q_options & (1 << 20)) > 0)
+    if q_land_option or q_RTL_option then
+        return false
+    end
+    return true -- always ok if not a QuadPlane
+end
+
+-- A single check to ensure AIRSPEED_STALL < AIRSPEED_MIN < AIRSPEED_CRUISE < AIRSPEED_MAX
+local function airspeed_check()
+    local airspeed_stall = AIRSPEED_STALL:get() or -1
+    local airspeed_min = AIRSPEED_MIN:get() or -1
+    local airspeed_cruise = AIRSPEED_CRUISE:get() or -1
+    local airspeed_max = AIRSPEED_MAX:get() or -1
+
+    if airspeed_stall > 0 then
+        if airspeed_min < airspeed_stall then
+            return false
+        end
+    end
+    if airspeed_cruise < airspeed_min then
+        return false
+    end
+    if airspeed_max < airspeed_cruise then
+        return false
+    end
```

Comment:
I thought we had plane checks for this stuff, if not then this might be worth adding to the c++.

Reply:
No current checks for this, interesting that this throws errors with default parameters (specifically SCALING_SPEED). could be moved to c++ later I guess.